pkg/securitypolicy/securitypolicyenforcer_rego.go

```diff
@@ -301,40 +316,53 @@ func errorString(errors interface{}) string {
 	return strings.Join(output, ",")
 }
 
-func (policy *regoEnforcer) getReasonNotAllowed(enforcementPoint string, input inputData) error {
-	inputJSON, err := json.Marshal(policy.redactSensitiveData(input))
+func (policy *regoEnforcer) denyWithError(ctx context.Context, policyError error, input inputData) error {
+	policyDecision := map[string]interface{}{
+		"input":       input,
+		"decision":    "deny",
+		"reason":      invalidError,
+		"policyError": policyError.Error(),
+	}
+
+	errorJson, err := json.Marshal(policyDecision)
 	if err != nil {
-		return fmt.Errorf("%s not allowed by policy. Input unavailable due to marshalling error", enforcementPoint)
+		log.G(ctx).Errorf("Unable to marshal error object: %v", err)
+		errorJson = []byte(`"Unable to marshal error object"`)
 	}
 
-	defaultMessage := fmt.Errorf("%s not allowed by policy. Security policy is not valid. Please check security policy or re-generate with tooling. Input: %s", enforcementPoint, string(inputJSON))
+	log.G(ctx).WithField("error", string(errorJson)).Debug("Policy decision")
```

Comment:
can the error json contain any sensitive data? if so, should we also redact it somehow?

Reply:
Shouldn't be a concern here, as only the input has sensitive data in it.